src/dataflow/source/kafka.rs

```diff
@@ -371,15 +379,29 @@ fn update_partition_list(
         if next_partition_ts.get(&partition_id).is_none() {
             next_partition_ts.insert(partition_id.clone(), closed_ts);
         }
-        // The last processed offset is always 0
         if last_processed_offsets.get(&partition_id).is_none() {
-            last_processed_offsets.insert(partition_id.clone(), 0);
+            last_processed_offsets.insert(partition_id.clone(), start_offset);
```

Comment:
I don't think this is necessarily correct when a new partition is added. If you start at offset 2100 and then add a new partition, new records will be added for that partition at offsets 0,1,2,3, etc. I would suggest that we explicitly disallow new partitions and ignore them when start_offset > 0 for now.

Reply:
Probably the best and most flexible thing to do is have per-partition start offsets, as you suggest.